### PR TITLE
GITHUB: Adding PR template

### PR DESCRIPTION
**Pull request by shamisp**
PR template proposal. Feel free to comment.

Signed-off-by: Pavel Shamis (Pasha) <pasharesearch@gmail.com>

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## What
+_Describe what this PR is doing._ 
+
+## Why ?
+_Justification for the PR. If there is existing issue/bug please reference. For
+bug fixes why and what can be merged in a single item._
+
+## How ?
+_It is optional but for complex PRs please provide information about the design,
+architecture, approach, etc.
```
Comment (miked-mellanox) [inline PULL_REQUEST_TEMPLATE.md]: ## How it was tested

## todo/limitations (if applicable)

## Performance before/after (if applicable)

maybe worth adding "checklist" as well, smth like:

## checklist

- [] for major feature: add wiki page with usage example and description 
- []

Comment (miked-mellanox) [reply PULL_REQUEST_TEMPLATE.md]: WDYT

Comment (shamisp) [reply PULL_REQUEST_TEMPLATE.md]: My goal was to keep it as simple as possible. Describe in layman's terms what you are doing.
Later on we can extend this. As for now I would love to see clear description why we introduce a change.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7200/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7201/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
@yosefe ?

**Comment by yosefe** [comment]
I prefer to avoid PR templates and keep the info in commit messages, since it's easier to look at revision history this way.

**Comment by shamisp** [comment]
Commit message should provide enough information about the patch.
PR should be more like RFC - include information and extended justification.
We have not been doing very good job in either one.

This PR template as a reminder for developer. I do think it is useful
